### PR TITLE
Ops 1338 refactor filter

### PR DESCRIPTION
**Pull request by johndeange**
## What changed

Refactor Filter component to improve reusability.

## Issue

https://github.com/HHS/OPRE-OPS/issues/1338

## How to test

Automated tests pass.

### CODE REVIEW

File: frontend/src/components/UI/FilterButton/FilterButton.jsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    const [showModal, setShowModal] = React.useState(false);
+
+    const handleApplyFilter = () => {
+        applyFilter();
+        setShowModal(false);
+    };
+
+    const handleResetFilter = () => {
+        resetFilter();
+        // setShowModal(false);
```
Comment (fpigeonjr) [inline frontend/src/components/UI/FilterButton/FilterButton.jsx:24]: commented code here

File: frontend/src/components/UI/FilterButton/FilterButton.jsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    };
+
+    Modal.setAppElement("#root");
+
+    return (
+        <div className={customStyles.container} id="filter-container">
+            <button
+                className={`usa-button display-flex flex-align-center ${customStyles.filterButton} margin-right-0`}
```
Comment (fpigeonjr) [inline frontend/src/components/UI/FilterButton/FilterButton.jsx:32]: prob want custom styles at the end of the styles class for overrides

File: frontend/src/components/UI/FilterButton/FilterButton.module.css [hunk 3]
```diff
@@ -0,0 +1,31 @@
+.filterButton {
+  background-color: #162E51;
+  color: #FFFFFF;
+}
+
+.filterModal {
+  position: absolute;
+  width: 460px;
```
Comment (fpigeonjr) [inline frontend/src/components/UI/FilterButton/FilterButton.module.css:8]: should we use rem units?

File: frontend/src/components/UI/FilterTags/FilterTags.jsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+ * @param {Object} props - The component props.
+ * @param {Function} props.removeFilter - A function to call to remove a filter/tag.
+ * @param {array} props.tagsList - An array of tags to display.
+ * @returns {JSX.Element} - The procurement shop select element.
+ */
+export const FilterTags = ({ removeFilter, tagsList }) => {
+    const FilterTag = ({ tag }) => (
+        <div
+            className="font-12px height-205 radius-md bg-brand-primary-light display-flex flex-align-center"
+            style={{ width: "fit-content", padding: "5px" }}
```
Comment (fpigeonjr) [inline frontend/src/components/UI/FilterTags/FilterTags.jsx:14]: we have a helper class for `fit-content` called: 
`.width-fit-content` in the `/sass/uswds/styles.scss`

File: frontend/src/pages/agreements/list/AgreementsFilterButton.module.css [hunk 5]
```diff
@@ -1,33 +1,7 @@
...
-  right: 0;
-  top: 3em;
-  z-index: 10;
-  border-radius: 4px;
-  border: 1px solid #A9AEB1;
-  /*background: #F0F0F0;*/
-  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.15);
-}
-
 .modalBackgroundColor {
```
Comment (fpigeonjr) [inline frontend/src/pages/agreements/list/AgreementsFilterButton.module.css:1]: could use `bg-brand-neutral-lightest` class from the `/sass/uswds/styles.scss`

File: frontend/src/pages/agreements/list/AgreementsFilterButton.module.css [hunk 6]
```diff
@@ -1,33 +1,7 @@
...
-.agreementsFilterOverlay {
-  background-color: #A9AEB1;
-}
-
-.container {
-  position: relative;
-}
-
 .legendColor {
   color: #565C65;
```
Comment (fpigeonjr) [inline frontend/src/pages/agreements/list/AgreementsFilterButton.module.css:6]: should we add this into our colors helpers?

File: frontend/src/pages/agreements/list/AgreementsFilterSection.jsx [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+/**
+ * A header section of the agreements page that contains the filters.
+ * @param {Object} props - The component props.
+ * @param {Object} props.filters - The current filters.
+ * @param {Function} props.setFilters - A function to call to set the filters.
+ * @returns {JSX.Element} - The procurement shop select element.
+ */
+export const AgreementsFilterSection = ({ filters, setFilters }) => {
+    return (
+        <div className="padding-top-05 padding-bottom-05 display-flex" style={{ justifyContent: "space-between" }}>
```
Comment (fpigeonjr) [inline frontend/src/pages/agreements/list/AgreementsFilterSection.jsx:14]: may be helpful to have this as a custom class in our styles helper file called something like `flex-justify-space-between`

File: frontend/src/components/UI/FilterButton/utils.js [hunk 8]
```diff
@@ -0,0 +1,22 @@
+import _ from "lodash";
+
+const setFilterList = (prevState, filterKeyString, stateObject, onlyAllowOne = false) => {
```
Comment (fpigeonjr) [inline frontend/src/components/UI/FilterButton/utils.js:3]: some black magic here 🪄

**Review by fpigeonjr — APPROVED** [review]
Looks good @johndeange, have some non-blocking suggestions here:

File: frontend/src/pages/agreements/list/AgreementsFilterButton.module.css (hunk 6 above)
Comment (johndeange) [reply frontend/src/pages/agreements/list/AgreementsFilterButton.module.css:6]: this css was removed

**Review by xlorepdarkhelm — APPROVED** [review]

**Review by tdonaworth — APPROVED** [review]

**Pull request merged**